pkg/apis/query/v0alpha1/query.go

```diff
@@ -46,6 +48,9 @@ type GenericDataQuery struct {
 	// RefID is the unique identifier of the query, set by the frontend call.
 	RefID string `json:"refId"`
 
+	// Optionally specify the required result type and version
+	ResponseContract *ResultDataContract `json:"resultDataContract,omitempty"`
```

Comment:
thoughts on better JSON names?  it needs to be slightly obscure since this lives in the same level as all existing datasource queries -- "format" and "type" etc are already widely used.

Maybe we can do an undersscore prefix for internal things? "_format"?

Reply:
`ResponseAssertions`